src/main/java/com/sprint/mission/discodeit/controller/AuthController.java

```diff
@@ -1,32 +1,65 @@
 package com.sprint.mission.discodeit.controller;
 
+import com.sprint.mission.discodeit.controller.api.AuthApi;
+import com.sprint.mission.discodeit.dto.data.DiscodeitUserDetails;
 import com.sprint.mission.discodeit.dto.data.UserDto;
-import com.sprint.mission.discodeit.dto.request.LoginRequest;
-import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.dto.request.UserRoleUpdateRequest;
+import com.sprint.mission.discodeit.exception.user.InvalidCredentialsException;
 import com.sprint.mission.discodeit.service.AuthService;
-import jakarta.validation.Valid;
+import com.sprint.mission.discodeit.service.UserService;
+import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.csrf.CsrfToken;
+import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.RestController;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/api/auth")
-public class AuthController {
+public class AuthController implements AuthApi {
 
   private final AuthService authService;
+  private final UserService userService;
 
-  @PostMapping("login")
-  public ResponseEntity<UserDto> login(@RequestBody @Valid LoginRequest loginRequest) {
-    UserDto user = authService.login(loginRequest);
-    return ResponseEntity
-        .status(HttpStatus.OK)
-        .body(user);
+//  @PostMapping(path = "login")
+//  public ResponseEntity<UserDto> login(@RequestBody @Valid LoginRequest loginRequest) {
+//    log.info("로그인 요청: username={}", loginRequest.username());
+//    UserDto user = authService.login(loginRequest);
+//    log.debug("로그인 응답: {}", user);
+//    return ResponseEntity
+//        .status(HttpStatus.OK)
+//        .body(user);
+//  }
+
+  @GetMapping("/csrf-token")
+  public ResponseEntity<Void> getCsrfToken(CsrfToken csrfToken) {
+    String token = csrfToken.getToken();
+    log.debug("Csrf token 요청 : {}", token);
+    return ResponseEntity.status(HttpStatus.OK).body(null);
+  }
+
+  @GetMapping("/me")
+  public ResponseEntity<UserDto> me() {
```

Comment:
`@AuthenticationPrincipal`을 사용하면 SecurityContextHolder에 직접 접근하는 로직을 없앨 수 있어요!
```suggestion
  public ResponseEntity<UserDto> me(@AuthenticationPrincipal DiscodeitUserDetails userDetails) {
```